signbank/dictionary/csv_import.py

```diff
@@ -231,10 +231,23 @@ def import_nzsl_share_gloss_csv(request):
         for row in glossreader:
             if glossreader.line_num == 1:
                 continue
-            if row["id"] not in existing_nzsl_share_ids:
-                new_glosses.append(row)
+            if row["id"] in existing_nzsl_share_ids:
+
+                # nzsl_share_id is not a reliable index, due to manual intervention
+                try:
+                    gloss = Gloss.objects.filter(nzsl_share_id=row["id"]).get()
```

Comment:
issue: this feels like we might be doubling up work

Above, we get `existing_nzsl_share_ids` before we enter the loop, which looks very similar to what you're doing here:

```
        existing_nzsl_share_ids = set(
            Gloss.objects.exclude(nzsl_share_id__exact="").values_list(
                "nzsl_share_id", flat=True
            )
        )
```

Should we maybe instead be changing that query to return more of the gloss and leveraging that here?

Reply:
I did think of that - eg. a pre-fetch - but given that the import is quite fast even for several hundred glosses it seemed a risky and unnecessary optimisation.

Reply:
One could definitely go through and try out such optimisations but there would be considerable debug.
Given the performance is fine, I'm ignoring it for the purposes of this ticket.